scenarios/22_North_Pole.cfg

```diff
@@ -327,46 +370,8 @@
         [/filter]
 
         {MODIFY_UNIT side=1 upkeep loyal}
-    [/event]
-
-    #make summoned ground troops move faster on snow
-    [event]
-        name=recruit,recall,post advance
-        first_time_only=no
-
-        [filter]
-            side=1
-            [not]
-                type=EoMa_Dimensional_Gate,EoMa_Dimensional_Gate_II,EoMa_Jinn,EoMa_Carpet_Rider,EoMa_Great_Jinn,EoMa_Wonderful_Jinn,EoMa_Mystical_Jinn,EoMa_Efreet,EoMa_Great_Efreeti,EoMa_Carpet_Master,EoMa_Air_Elemental,EoMa_Air_Avatar,EoMa_Air_God,EoMa_Fire_Avatar,EoMa_Fire_God
-            [/not]
-        [/filter]
-
-        {MODIFY_UNIT (x,y=$x1,$y1) movement_costs.frozen 2}
-        [object]
-            [filter]
-                x,y=$x1,$y1
-            [/filter]
-            [effect]
-                apply_to=defense
-                replace=yes
-                [defense]
-                    frozen=60
-                [/defense]
-            [/effect]
-            duration=scenario
-            silent=yes
-        [/object]
-    [/event]
-    [event]
-        name=recruit,recall,post advance
-        first_time_only=no
-
-        [filter]
-            side=1
-            type=EoMa_Dimensional_Gate,EoMa_Dimensional_Gate_II,EoMa_Jinn,EoMa_Carpet_Rider,EoMa_Great_Jinn,EoMa_Wonderful_Jinn,EoMa_Mystical_Jinn,EoMa_Efreet,EoMa_Great_Efreeti,EoMa_Carpet_Master,EoMa_Air_Elemental,EoMa_Air_Avatar,EoMa_Air_God,EoMa_Fire_Avatar,EoMa_Fire_God
-        [/filter]
-
-        {MODIFY_UNIT (x,y=$x1,$y1) movement_costs.frozen 1}
+        [allow_undo]
```

Comment:
I am not convinced all the events where you allow undo are actually safe to undo

Reply:
Well, ``[allow_undo]`` has no effect for events for which it's known that cannot be undone (e.g. ``post advance``). I've played around with undoing actions in this scenario plenty, and I have only managed to trigger OOS when attempting to undo a recall event that called ``{ADVANCE_UNIT}``, so these should be safe.